presto-native-execution/presto_cpp/main/types/tests/FunctionMetadataTest.cpp

```diff
@@ -0,0 +1,156 @@
+    EXPECT_EQ(j.at("outputType"), "bigint");
+    auto paramTypes = j.at("paramTypes");
+    VELOX_CHECK(
+        ((paramTypes == std::vector<std::string>({})) ||
+         (paramTypes == std::vector<std::string>({"T"}))));
+
+    auto routineCharacteristics = j.at("routineCharacteristics");
+    EXPECT_EQ(
+        routineCharacteristics.at("language"), protocol::Language({"CPP"}));
+    EXPECT_EQ(
+        routineCharacteristics.at("determinism"),
+        protocol::Determinism::DETERMINISTIC);
+    EXPECT_EQ(
+        routineCharacteristics.at("nullCallClause"),
+        protocol::NullCallClause::CALLED_ON_NULL_INPUT);
+
+    auto aggregateMetadata = j.at("aggregateMetadata");
+    EXPECT_EQ(aggregateMetadata.at("intermediateType"), "bigint");
+    EXPECT_EQ(aggregateMetadata.at("isOrderSensitive"), false);
+  }
+}
+
+TEST(FunctionMetadataTest, testSum) {
+  aggregate::prestosql::registerAllAggregateFunctions("");
+  std::string functionName = "sum";
+  json metadataList = getJsonMetadataForFunction(functionName);
+  const auto numSignatures = 7;
+  EXPECT_EQ(metadataList.size(), numSignatures);
+
+  for (auto i = 0; i < numSignatures; i++) {
+    json j = metadataList.at(i);
+    EXPECT_EQ(j.at("functionKind"), protocol::FunctionKind::AGGREGATE);
+    EXPECT_EQ(j.at("docString"), functionName);
+    EXPECT_EQ(j.at("schema"), "default");
+    auto outputType = j.at("outputType");
+    VELOX_CHECK(
+        ((outputType == "real") || (outputType == "double") ||
+         (outputType == "DECIMAL(38,a_scale)") || (outputType == "bigint")));
+    auto paramTypes = j.at("paramTypes");
+    VELOX_CHECK(
+        ((paramTypes == std::vector<std::string>({"real"})) ||
+         (paramTypes == std::vector<std::string>({"double"})) ||
+         (paramTypes ==
+          std::vector<std::string>({"DECIMAL(a_precision,a_scale)"})) ||
+         (paramTypes == std::vector<std::string>({"tinyint"})) ||
+         (paramTypes == std::vector<std::string>({"smallint"})) ||
+         (paramTypes == std::vector<std::string>({"integer"})) ||
+         (paramTypes == std::vector<std::string>({"bigint"}))));
+
+    auto routineCharacteristics = j.at("routineCharacteristics");
+    EXPECT_EQ(
+        routineCharacteristics.at("language"), protocol::Language({"CPP"}));
+    EXPECT_EQ(
+        routineCharacteristics.at("determinism"),
+        protocol::Determinism::DETERMINISTIC);
+    EXPECT_EQ(
+        routineCharacteristics.at("nullCallClause"),
+        protocol::NullCallClause::CALLED_ON_NULL_INPUT);
+
+    auto aggregateMetadata = j.at("aggregateMetadata");
+    auto intermediateType = aggregateMetadata.at("intermediateType");
+    VELOX_CHECK(
+        (intermediateType == "double") || (intermediateType == "VARBINARY") ||
+        (intermediateType == "bigint"));
+    EXPECT_EQ(aggregateMetadata.at("isOrderSensitive"), false);
+  }
+}
+
+TEST(FunctionMetadataTest, testRank) {
+  window::prestosql::registerAllWindowFunctions("");
```

Comment:
Are aggregates also available in window function list ? If yes, then we should check an aggregate function here as well.

Reply:
No aggregates are not included in the window function list since the function `getSortedWindowNames` removes aggregates. Will update the function in the velox PR to add aggregates, thanks for pointing it out.